onward/app/controllers/OnwardResponseController.scala

```diff
@@ -0,0 +1,60 @@
+package controllers
+
+import common.{Edition, ImplicitControllerExecutionContext, JsonComponent}
+import feed.MostReadAgent
+import model.{ApplicationContext, Cached}
+import model.dotcomrendering.OnwardCollectionResponse
+import play.api.libs.ws.WSClient
+import play.api.mvc.{Action, AnyContent, BaseController, ControllerComponents, RequestHeader, Result}
+import play.twirl.api.Html
+import renderers.DotcomRenderingService
+import services.PopularInTagService
+
+import scala.concurrent.Future
+
+class OnwardResponseController(
+    ws: WSClient,
+    remoteRenderer: DotcomRenderingService,
+    popularInTagService: PopularInTagService,
+    mostReadAgent: MostReadAgent,
+    val controllerComponents: ControllerComponents,
+)(implicit context: ApplicationContext)
+    extends BaseController
+    with ImplicitControllerExecutionContext {
+
+  def popularInTag(tag: String)(implicit request: RequestHeader): Future[OnwardCollectionResponse] = {
+    val edition = Edition(request)
+    val excludeTags = request.queryString.getOrElse("exclude-tag", Nil)
+    val itemViewCounts = mostReadAgent.getViewCounts
+
+    popularInTagService.fetch(edition, tag, excludeTags, itemViewCounts)
+  }
+  def popularInTagJson(tag: String): Action[AnyContent] =
+    Action.async { implicit request =>
+      popularInTag(tag) flatMap renderJson
+    }
+  def popularInTagHtml(tag: String): Action[AnyContent] =
+    Action.async { implicit request =>
+      popularInTag(tag) flatMap renderHtml
+    }
+
+  private def renderJson(
+      onwardsCollection: OnwardCollectionResponse,
+  )(implicit request: RequestHeader): Future[Result] = {
+    Future.successful(Cached(600) {
+      JsonComponent
+        .fromWritable[OnwardCollectionResponse](onwardsCollection)(
+          request,
+          OnwardCollectionResponse.collectionWrites,
+        )
+    })
+  }
+
+  private def renderHtml(
+      onwardsCollection: OnwardCollectionResponse,
+  )(implicit request: RequestHeader): Future[Result] = {
+    remoteRenderer.getOnward(ws, onwardsCollection) map { result =>
+      Cached(600)(JsonComponent("html" -> new Html(result)))
```

Comment:
I've always thought that the pattern of wrapping a HTML blob by itself in JSON interesting! We don't actually do it everywhere, that I know of football embeds and the "Show more" fixture endpoints return the raw HTML to be put on a page. I personally quite like just returning the 
1. It simplifies DotcomRenderingService as you don't need to use the postWithoutHandler
2. You can test the endpoint in the browser and easily verify that everything looks Okay

Reply:
This makes a lot of sense, but there seems to be more examples using the `{"html": string}` pattern, so I might leave it as that for the time being.